operator/standalone/update.go

```diff
@@ -0,0 +1,386 @@
+				pipeline.NewStepFromFunc("apply changes to helm release", u.applyValuesFromInstance),
+				pipeline.NewStepFromFunc("ensure helm release", u.ensureHelmRelease),
+			),
+
+			pipeline.NewStepFromFunc("update status condition", u.markInstanceAsProgressing),
+		).
+		RunWithContext(ctx).Err()
+}
+
+// WaitUntilAllResourceReady runs a pipeline that verifies if all dependent resources are ready.
+// It will add the conditions.TypeReady condition to the status field (and update it) if it's considered ready.
+// No error is returned in case the instance is not considered ready.
+func (u *UpdateStandalonePipeline) WaitUntilAllResourceReady(ctx context.Context) error {
+	return pipeline.NewPipeline().
+		WithSteps(
+			pipeline.NewStepFromFunc("fetch helm release", u.fetchHelmRelease),
+			pipeline.If(pipeline.And(u.isHelmReleaseReady, u.isPodRunning),
+				pipeline.NewPipeline().WithNestedSteps("finish updating",
+					pipeline.NewStepFromFunc("mark instance ready", u.markInstanceAsReady),
+				),
+			),
+		).
+		RunWithContext(ctx).Err()
+}
+
+// fetchHelmRelease fetches the Helm release for the given instance.
+func (u *UpdateStandalonePipeline) fetchHelmRelease(ctx context.Context) error {
+	helmRelease := &helmv1beta1.Release{}
+	err := u.client.Get(ctx, client.ObjectKey{Name: u.instance.Status.HelmChart.DeploymentNamespace}, helmRelease)
+	u.helmRelease = helmRelease
+	return err
+}
+
+// useReleaseValues unmarshalls and saves release values to be used in subsequent steps
+//
+// This step assumes that the release has been fetched first via fetchHelmRelease.
+func (u *UpdateStandalonePipeline) useReleaseValues(_ context.Context) error {
+	values := helmvalues.V{}
+	err := helmvalues.Unmarshal(u.helmRelease.Spec.ForProvider.Values, &values)
+	u.helmValues = values
+	return err
+}
+
+// applyValuesFromInstance merges the user-defined and -exposed Helm values into the current Helm values map.
+func (u *UpdateStandalonePipeline) applyValuesFromInstance(_ context.Context) error {
+	resources := helmvalues.V{
+		"auth": helmvalues.V{
+			"enablePostgresUser": u.instance.Spec.Parameters.EnableSuperUser,
+			"existingSecret":     getCredentialSecretName(),
+			"database":           u.instance.Name,
+			"username":           u.instance.Name,
+		},
+		"primary": helmvalues.V{
+			"resources": helmvalues.V{
+				"limits": helmvalues.V{
+					"memory": u.instance.Spec.Parameters.Resources.MemoryLimit.String(),
+				},
+			},
+			"persistence": helmvalues.V{
+				"size": u.instance.Spec.Parameters.Resources.StorageCapacity.String(),
+			},
+		},
+		"fullnameOverride": getDeploymentName(),
+		"networkPolicy": helmvalues.V{
+			"enabled": true,
+			"ingressRules": helmvalues.V{
+				"primaryAccessOnlyFrom": helmvalues.V{
+					"enabled": true,
+					"namespaceSelector": helmvalues.V{
+						"kubernetes.io/metadata.name": u.instance.Namespace,
+					},
+				},
+			},
+		},
+	}
+	helmvalues.Merge(resources, &u.helmValues)
+	return nil
+}
```

Comment:
Can we re-use the same values as in the Create pipeline? to avoid duplication and bugs...
Maybe the helmvalue.V{...} can be extracted into a function that works for both the Create and Update pipeline.

Or is this intended in a follow-up PR for the deduplication refactoring?